packages/mco/components/disaster-recovery/applications/failover-and-relocate-modal/peer-status.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import { getName } from '@odf/shared/selectors';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import {
+  StatusIconAndText,
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'i18next';
+import { Flex, FlexItem } from '@patternfly/react-core';
+import { UnknownIcon } from '@patternfly/react-icons';
+import { PEER_READINESS } from '../../../../constants';
+import { findPeerStatus } from '../../../../utils';
+import {
+  ACTION_TYPE,
+  FailoverAndRelocateState,
+  FailoverAndRelocateAction,
+  FailoverAndRelocateType,
+} from './reducer';
+
+const initalPeerStatus = (t: TFunction) => ({
+  peerReadiness: {
+    icon: <UnknownIcon />,
+    text: PEER_READINESS(t).UNKNOWN,
+  },
+  lastHealthCheck: {
+    icon: <UnknownIcon />,
+    text: t('Unknown'),
+  },
+  lastSnapshotTakenOn: {
+    icon: <UnknownIcon />,
+    text: t('Unknown'),
+  },
+});
+
+type StatusProps = {
+  icon: JSX.Element;
+  text: string;
+};
+
+type StatusType = {
+  peerReadiness: StatusProps;
+  lastHealthCheck: StatusProps;
+  lastSnapshotTakenOn: StatusProps;
+};
+
+export const PeerStatus: React.FC<PeerStatusProps> = ({
+  action,
+  state,
+  dispatch,
+}) => {
+  const { t } = useCustomTranslation();
+  const [peerStatus, setPeerStatus] = React.useState<StatusType>(
+    initalPeerStatus(t)
+  );
+  const setErrorMessage = React.useCallback(
+    (errorMessage: string) => {
+      dispatch({
+        type: FailoverAndRelocateType.SET_ERROR_MESSAGE,
+        payload: {
+          peerStatusErrorMessage: errorMessage,
+        },
+      });
+    },
+    [dispatch]
+  );
+
+  React.useEffect(() => {
+    if (!!state.selectedSubsGroups.length) {
+      const status = state.drPolicyControlState?.reduce((obj, acmToDRState) => {
+        if (
+          state.selectedSubsGroups.includes(
+            getName(acmToDRState?.drPolicyControl)
+          )
+        ) {
+          const peerReadiness =
+            obj.peerReadiness.text !== PEER_READINESS(t).PEER_NOT_READY
+              ? findPeerStatus(acmToDRState?.drPolicyControl)
+                ? PEER_READINESS(t).PEER_READY
+                : PEER_READINESS(t).PEER_NOT_READY
+              : obj.peerReadiness.text;
+          return {
+            ...obj,
+            peerReadiness: {
+              text: peerReadiness,
+              icon:
+                peerReadiness === PEER_READINESS(t).PEER_READY ? (
```

Comment:
```suggestion
                peerReadiness === PEER_READINESS(t).PEER_READY ? (
```
I don't think this a right approach where we need to resolve translations even for performing string comparisons.  We should have intermediate values and use translations only on places that are visible to the end user.